st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -144,6 +149,9 @@ def connect(self):
                  '_username': self.username, '_timeout': self.timeout}
         self.logger.debug('Connecting to server', extra=extra)
 
+        if self.bastion_host and self._bastion_connect():
+            conninfo['sock'] = self.bastion_socket
```

Comment:
I saw your comment but I am not sure how this really works. We should be connecting to the end host and running the command on the end host. This means proxying needs to happen on the bastion host. I saw your test cases and it looks like your tests cover only localhost. Maybe you can test this with hosts != localhost as end hosts? I'll look at paramiko too to see if they support bastion directly.